library/std/src/env.rs

```diff
@@ -318,20 +318,18 @@ impl Error for VarError {
 ///
 /// # Safety
 ///
-/// Even though this function is currently not marked as `unsafe`, it needs to
-/// be because invoking it can cause undefined behaviour. The function will be
-/// marked `unsafe` in a future version of Rust. This is tracked in
-/// [rust#27970](https://github.com/rust-lang/rust/issues/27970).
-///
 /// This function is safe to call in a single-threaded program.
 ///
-/// In multi-threaded programs, you must ensure that are no other threads
-/// concurrently writing or *reading*(!) from the environment through functions
-/// other than the ones in this module. You are responsible for figuring out
-/// how to achieve this, but we strongly suggest not using `set_var` or
-/// `remove_var` in multi-threaded programs at all.
+/// This function is also always safe to call on Windows, in single-threaded
+/// and multi-threaded programs.
 ///
-/// Most C libraries, including libc itself do not advertise which functions
+/// In multi-threaded programs on other operating systems, you must ensure that
+/// are no other threads concurrently writing or *reading*(!) from the
+/// environment through functions other than the ones in this module. You are
+/// responsible for figuring out how to achieve this, but we strongly suggest
+/// not using `set_var` or `remove_var` in multi-threaded programs at all.
+///
+/// Most C libraries, including libc itself, do not advertise which functions
 /// read from the environment. Even functions from the Rust standard library do
 /// that, e.g. for DNS lookups from [`std::net::ToSocketAddrs`].
```

Comment:
This should probably be part of the previous paragraph? It's part of "what needs to be done to ensure that no concurrent threads read or write the environment".

Reply:
Hmm, I think the paragraph is long enough as-is. I'd say it's okay since this gives a wider view of the ecosystem.

Reply:
It should be one paragraph, one point. Currently this paragraph sits here on its own in a somewhat disconnected way to the rest of the document, which is confusing. "Why are you telling me this? Oh, it's part of that argument in the middle of the previous paragraph..."

Reply:
Done.